src/utils.jl

```diff
@@ -5,24 +5,31 @@ const Tracked = 10
 const Derived = 11
 export Tracked, Derived
 
-function unsafe_to_llvm(val, ctx)
-    T_jlvalue = LLVM.StructType(LLVM.LLVMType[]; ctx)
+function unsafe_to_llvm(val)
+    T_jlvalue = LLVM.StructType(LLVM.LLVMType[])
     T_prjlvalue = LLVM.PointerType(T_jlvalue, Tracked)
     T_prjlvalue_UT = LLVM.PointerType(T_jlvalue)
     fill_val = unsafe_to_pointer(val)
-    fill_val = LLVM.ConstantInt(convert(UInt, fill_val); ctx)
+    fill_val = LLVM.ConstantInt(convert(UInt, fill_val))
     fill_val = LLVM.const_inttoptr(fill_val, T_prjlvalue_UT)
     LLVM.const_addrspacecast(fill_val, T_prjlvalue)
 end
+
 export unsafe_to_llvm
 
-function makeInstanceOf(@nospecialize(T), ctx)
-    @assert Core.Compiler.isconstType(T)
-    @assert T <: Type
-    return unsafe_to_llvm(T.parameters[1], ctx)
+import LLVM: context!, ts_context!, context
+
+@static if VERSION < v"1.9-"
+    context(ctx::LLVM.Context) = ctx
+else
+    context!(f, ctx::LLVM.ThreadSafeContext) = LLVM.ts_context!(f, ctx)
 end
 
-export makeInstanceOf
+@static if VERSION < v"1.9-"
+    has_context() = LLVM._has_context()
+else
+    has_context() = LLVM._has_ts_context() ###??
```

Comment:
@vchuravy @maleadt 

separately it would be nice if we can refactor to not need this